### DISCUSSION TITLE
how to call `train_one_epoch` explicitly?

### DISCUSSION THREAD

**Discussion by marsggbo** [Lightning-AI/pytorch-lightning#13289] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
Pytorch-lightning seems to provide only `fit`, `validate`, `test`, and `predict`. How can we explicitly call a function of training only one epoch? Below is a pseudo-code

```
for epoch in range(10):
    if epoch <= 5:
        # do something
    else:
        # do something
    train_results = trainer.train_one_epoch(...)
    valid_results = trainer.validate(....)
```

**Comment by akihironitta** [comment]
@marsggbo Have you tried overriding hooks? PyTorch Lightning provides quite a number of hooks so that you can customise the default behaviour:
https://pytorch-lightning.readthedocs.io/en/stable/common/lightning_module.html#hooks

You can also utilise `current_epoch` property of the trainer if you want to write some logic that depends on epoch index:
https://pytorch-lightning.readthedocs.io/en/stable/common/trainer.html#current-epoch
